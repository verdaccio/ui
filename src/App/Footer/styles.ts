import styled from '@emotion/styled';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

export const Wrapper = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  background: theme?.palette.type === 'light' ? theme?.palette.snow : theme?.palette.cyanBlue,
  borderTop: `1px solid ${theme?.palette.greyGainsboro}`,
  color: theme?.palette.type === 'dark' ? theme?.palette.white : theme?.palette.nobel01,
  fontSize: '14px',
  padding: '20px',
}));

export const Inner = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-end',
  width: '100%',
  [`@media (min-width: ${theme?.breakPoints.medium}px)`]: {
    minWidth: 400,
    maxWidth: 800,
    margin: 'auto',
    justifyContent: 'space-between',
  },
  [`@media (min-width: ${theme?.breakPoints.large}px)`]: {
    maxWidth: 1240,
  },
}));

export const Left = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  alignItems: 'center',
  display: 'none',
  [`@media (min-width: ${theme?.breakPoints.medium}px)`]: {
    display: 'flex',
  },
}));

export const Right = styled(Left)({
  display: 'flex',
});

export const Love = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.love,
  padding: '0 5px',
}));
