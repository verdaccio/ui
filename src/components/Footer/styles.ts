import styled from '@emotion/styled';

import Icon from '../Icon/Icon';
import { Theme } from '../../design-tokens/theme';

export const Wrapper = styled('div')<{ theme?: Theme }>(props => ({
  background: props.theme && props.theme.palette.snow,
  borderTop: `1px solid ${props.theme && props.theme.palette.greyGainsboro}`,
  color: props.theme && props.theme.palette.nobel01,
  fontSize: '14px',
  padding: '20px',
}));

export const Inner = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-end',
  width: '100%',
  [`@media (min-width: ${theme && theme.breakPoints.medium}px)`]: {
    minWidth: 400,
    maxWidth: 800,
    margin: 'auto',
    justifyContent: 'space-between',
  },
  [`@media (min-width: ${theme && theme.breakPoints.large}px)`]: {
    maxWidth: 1240,
  },
}));

export const Left = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  alignItems: 'center',
  display: 'none',
  [`@media (min-width: ${theme && theme.breakPoints.medium}px)`]: {
    display: 'flex',
  },
}));

export const Right = styled(Left)({
  display: 'flex',
});

export const Earth = styled(Icon)({
  padding: '0 10px',
});

export const Flags = styled('span')<{ theme?: Theme }>(props => ({
  position: 'absolute',
  background: props.theme && props.theme.palette.greyAthens,
  padding: '1px 4px',
  borderRadius: 3,
  height: 20,
  display: 'inline-flex',
  alignItems: 'center',
  visibility: 'hidden',
  top: -2,
  ':before': {
    content: "''",
    position: 'absolute',
    top: '29%',
    left: -4,
    marginLeft: -5,
    border: '5px solid',
    borderColor: `${props.theme && props.theme.palette.greyAthens} transparent transparent transparent`,
    transform: 'rotate(90deg)',
  },
}));

export const ToolTip = styled('span')({
  position: 'relative',
  height: '18px',
  ':hover': {
    [`${Flags}`]: {
      visibility: 'visible',
    },
  },
});

export const Love = styled('span')<{ theme?: Theme }>(props => ({
  color: props.theme && props.theme.palette.love,
  padding: '0 5px',
}));

export const Flag = styled(Icon)({
  padding: '0 5px',
});

export const Logo = Flag;
