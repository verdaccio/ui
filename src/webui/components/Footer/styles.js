/**
 * @prettier
 * @flow
 */

import styled, { css } from 'react-emotion';
import mq from '../../utils/styles/media';
import Icon from '../Icon';
import colors from '../../utils/styles/colors';

export const Wrapper = styled.div`
  && {
    background: ${colors.snow};
    border-top: 1px solid ${colors.greyGainsboro};
    color: ${colors.nobel01};
    font-size: 14px;
    padding: 20px;
  }
`;

export const Inner = styled.div`
  && {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    width: 100%;
    ${mq.medium(css`
      min-width: 400px;
      max-width: 800px;
      margin: auto;
      justify-content: space-between;
    `)};
    ${mq.large(css`
      max-width: 1240px;
    `)};
  }
`;

export const Left = styled.div`
  && {
    align-items: center;
    display: none;
    ${mq.medium(css`
      display: flex;
    `)};
  }
`;

export const Right = styled(Left)`
  && {
    display: flex;
  }
`;

export const ToolTip = styled.span`
  && {
    position: relative;
    height: 18px;
  }
`;

export const Earth = styled(Icon)`
  && {
    padding: 0 10px;
  }
`;

export const Flags = styled.span`
  && {
    position: absolute;
    background: ${colors.greyAthens};
    padding: 1px 4px;
    border-radius: 3px;
    height: 20px;
    display: inline-flex;
    align-items: center;
    visibility: hidden;
    top: -2px;
    :before {
      content: '';
      position: absolute;
      top: 29%;
      left: -4px;
      margin-left: -5px;
      border: 5px solid;
      border-color: ${colors.greyAthens} transparent transparent transparent;
      transform: rotate(90deg);
    }
    ${ToolTip}:hover & {
      visibility: visible;
    }
  }
`;

export const Love = styled.span`
  && {
    color: ${colors.love};
    padding: 0 5px;
  }
`;

export const Flag = styled(Icon)`
  && {
    padding: 0 5px;
  }
`;

export const Logo = Flag;
