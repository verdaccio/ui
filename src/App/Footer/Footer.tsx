import React from 'react';
import { useTranslation, Trans } from 'react-i18next';

import { goToVerdaccioWebsite } from 'verdaccio-ui/utils/windows';

import { Wrapper, Left, Right, Earth, Flags, Love, Flag, Logo, Inner, ToolTip } from './styles';

/* eslint-disable react/jsx-key */
const Footer: React.FC = () => {
  const { t } = useTranslation();
  return (
    <Wrapper>
      <Inner>
        <Left>
          <Trans components={[<Love />]} i18nKey="footer.made-with-love-on" />
          <ToolTip>
            <Earth name="earth" size="md" />
            <Flags>
              <Flag name="spain" size="md" />
              <Flag name="nicaragua" size="md" />
              <Flag name="india" size="md" />
              <Flag name="brazil" size="md" />
              <Flag name="china" size="md" />
              <Flag name="austria" size="md" />
              <Flag name="germany" size="md" />
              <Flag name="taiwan" size="md" />
            </Flags>
          </ToolTip>
        </Left>
        <Right>
          {t('footer.powered-by')}
          <Logo img={true} name="verdaccio" onClick={goToVerdaccioWebsite} pointer={true} size="md" />
          {`/ ${window.VERDACCIO_VERSION}`}
        </Right>
      </Inner>
    </Wrapper>
  );
};

export default Footer;
