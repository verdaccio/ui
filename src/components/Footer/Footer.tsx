import React from 'react';

import { Wrapper, Left, Right, Earth, Flags, Love, Flag, Logo, Inner, ToolTip } from './styles';
import { goToVerdaccioWebsite } from '../../utils/windows';

const renderTooltip = (): JSX.Element => (
  <ToolTip>
    <Earth name="earth" size="md" />
    <Flags>
      <Flag name="spain" size="md" />
      <Flag name="nicaragua" size="md" />
      <Flag name="india" size="md" />
      <Flag name="brazil" size="md" />
      <Flag name="china" size="md" />
      <Flag name="austria" size="md" />
    </Flags>
  </ToolTip>
);
const POWERED_LABEL = 'Powered by';
const MADEWITH_LABEL = ' Made with';
const ON_LABEL = 'on';
const HEARTH_EMOJI = 'â™¥';

const renderRight = (version = window.VERDACCIO_VERSION): JSX.Element => {
  return (
    <Right>
      {POWERED_LABEL}
      <Logo img={true} name="verdaccio" onClick={goToVerdaccioWebsite} pointer={true} size="md" />
      {`/ ${version}`}
    </Right>
  );
};

const renderLeft = (): JSX.Element => (
  <Left>
    {MADEWITH_LABEL}
    <Love>{HEARTH_EMOJI}</Love>
    {ON_LABEL}
    {renderTooltip()}
  </Left>
);

const Footer: React.FC = () => (
  <Wrapper>
    <Inner>
      {renderLeft()}
      {renderRight()}
    </Inner>
  </Wrapper>
);

export default Footer;
