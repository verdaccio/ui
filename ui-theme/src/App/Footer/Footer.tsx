/* eslint-disable react/jsx-max-depth */

/* eslint-disable react/jsx-pascal-case */
import styled from '@emotion/styled';
import FlagsIcon from 'country-flag-icons/react/3x2';
import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Earth } from 'verdaccio-ui/components/Icons';
import Logo from 'verdaccio-ui/components/Logo';
import { Theme } from 'verdaccio-ui/design-tokens/theme';
import { useConfig } from 'verdaccio-ui/providers/config';
import { goToVerdaccioWebsite } from 'verdaccio-ui/utils/windows';

import { Inner, Left, Love, Right, Wrapper } from './styles';

/* eslint-disable react/jsx-key */
const Footer = () => {
  const { t } = useTranslation();
  const { configOptions } = useConfig();
  return (
    <Wrapper>
      <Inner>
        <Left>
          <Trans components={[<Love />]} i18nKey="footer.made-with-love-on" />
          <ToolTip>
            <StyledEarth />
            <Flags>
              <Icon>
                <FlagsIcon.ES />
              </Icon>
              <Icon>
                <FlagsIcon.NI />
              </Icon>
              <Icon>
                <FlagsIcon.IN />
              </Icon>
              <Icon>
                <FlagsIcon.BR />
              </Icon>
              <Icon>
                <FlagsIcon.CN />
              </Icon>
              <Icon>
                <FlagsIcon.AU />
              </Icon>
              <Icon>
                <FlagsIcon.DE />
              </Icon>
              <Icon>
                <FlagsIcon.TW />
              </Icon>
            </Flags>
          </ToolTip>
        </Left>
        <Right>
          {configOptions?.version && (
            <>
              {t('footer.powered-by')}
              <Logo onClick={goToVerdaccioWebsite} size="x-small" />
              {`/ ${configOptions.version}`}
            </>
          )}
        </Right>
      </Inner>
    </Wrapper>
  );
};

export default Footer;

const StyledEarth = styled(Earth)<{ theme?: Theme }>(({ theme }) => ({
  margin: theme.spacing(0, 1),
}));

const Flags = styled('span')<{ theme?: Theme }>(({ theme }) => ({
  display: 'inline-grid',
  gridTemplateColumns: 'repeat(8, max-content)',
  gridGap: theme.spacing(0, 1),
  position: 'absolute',
  background: theme?.palette.greyAthens,
  padding: '1px 4px',
  borderRadius: 3,
  height: 20,
  alignItems: 'center',
  visibility: 'hidden',
  top: -2,
  ':before': {
    content: "''",
    position: 'absolute',
    top: '29%',
    left: -4,
    marginLeft: -5,
    border: '5px solid',
    borderColor: `${theme?.palette.greyAthens} transparent transparent transparent`,
    transform: 'rotate(90deg)',
  },
}));

const Icon = styled('div')({
  width: '10px',
});

const ToolTip = styled('span')({
  position: 'relative',
  height: '18px',
  ':hover': {
    [`${Flags}`]: {
      visibility: 'visible',
    },
  },
});
