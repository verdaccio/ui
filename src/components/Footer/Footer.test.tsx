import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import Footer from './Footer';

jest.mock('../../../package.json', () => ({
  version: '4.0.0-alpha.3',
}));

describe('<Footer /> component', () => {
  let wrapper: ReactWrapper;
  beforeEach(() => {
    window.VERDACCIO_VERSION = 'v.1.0.0';
    wrapper = mount(<Footer />);
    delete window.VERDACCIO_VERSION;
  });

  test('should load the initial state of Footer component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
