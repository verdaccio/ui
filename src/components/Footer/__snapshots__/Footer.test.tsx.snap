// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component should load the initial state of Footer component 1`] = `
.emotion-41 {
  background: #f9f9f9;
  border-top: 1px solid #e3e3e3;
  color: #999999;
  font-size: 14px;
  padding: 20px;
}

.emotion-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

@media (min-width:768px) {
  .emotion-39 {
    min-width: 400px;
    max-width: 800px;
    margin: auto;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:1024px) {
  .emotion-39 {
    max-width: 1240px;
  }
}

.emotion-30 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
}

@media (min-width:768px) {
  .emotion-30 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-0 {
  color: #e25555;
  padding: 0 5px;
}

.emotion-28 {
  position: relative;
  height: 18px;
}

.emotion-28:hover .emotion-27 {
  visibility: visible;
}

.emotion-3 {
  box-sizing: initial;
  display: inline-block;
  cursor: default;
  width: 18px;
  height: 18px;
  padding: 0 10px;
}

.emotion-26 {
  position: absolute;
  background: #d3dddd;
  padding: 1px 4px;
  border-radius: 3px;
  height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  visibility: hidden;
  top: -2px;
}

.emotion-26:before {
  content: '';
  position: absolute;
  top: 29%;
  left: -4px;
  margin-left: -5px;
  border: 5px solid;
  border-color: #d3dddd transparent transparent transparent;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-6 {
  box-sizing: initial;
  display: inline-block;
  cursor: default;
  width: 18px;
  height: 18px;
  padding: 0 5px;
}

.emotion-37 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:768px) {
  .emotion-37 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-35 {
  box-sizing: initial;
  display: inline-block;
  cursor: pointer;
  width: 18px;
  height: 18px;
  padding: 0 5px;
}

.emotion-32 {
  width: 100%;
  height: auto;
}

<div
  class="emotion-41 emotion-42"
>
  <div
    class="emotion-39 emotion-40"
  >
    <div
      class="emotion-30 emotion-31"
    >
      Made with 
      <span
        class="emotion-0 emotion-1"
      >
        â™¥
      </span>
       on
      <span
        class="emotion-28 emotion-29"
      >
        <svg
          class="emotion-2 emotion-3 emotion-4"
        >
          <title>
            Earth
          </title>
          <use
            xlink:href="[object Object]#earth"
          />
        </svg>
        <span
          class="emotion-26 emotion-27"
        >
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              Spain
            </title>
            <use
              xlink:href="[object Object]#spain"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              Nicaragua
            </title>
            <use
              xlink:href="[object Object]#nicaragua"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              India
            </title>
            <use
              xlink:href="[object Object]#india"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              Brazil
            </title>
            <use
              xlink:href="[object Object]#brazil"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              China
            </title>
            <use
              xlink:href="[object Object]#china"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              Austria
            </title>
            <use
              xlink:href="[object Object]#austria"
            />
          </svg>
          <svg
            class="emotion-5 emotion-6 emotion-4"
          >
            <title>
              Germany
            </title>
            <use
              xlink:href="[object Object]#germany"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      class="emotion-37 emotion-38"
    >
      Powered by
      <span
        class="emotion-5 emotion-35 emotion-36"
        title="Verdaccio"
      >
        <img
          alt="Verdaccio"
          class="emotion-32 emotion-33"
          src="[object Object]"
        />
      </span>
      / v.1.0.0
    </div>
  </div>
</div>
`;
